src/io/flutter/sdk/FlutterSdk.java

```diff
@@ -61,6 +75,57 @@ public static FlutterSdk getFlutterSdk(@NotNull final Project project) {
     return project.isDisposed() ? null : getFlutterSdkByDartSdk(DartPlugin.getDartSdk(project));
   }
 
+  /**
+   * Returns the Flutter SDK for an unopened project that has a possibly broken "Dart SDK" project library.
+   * <p>
+   * (This can happen for a newly-cloned Flutter SDK where the Dart SDK is not cached yet.)
+   */
+  @Nullable
+  public static FlutterSdk getForProjectDir(final VirtualFile projectDir) {
+    final Library lib = loadDartSdkLibrary(projectDir);
+    if (lib == null) {
+      return null;
+    }
+
+
+    final String[] urls = lib.getUrls(OrderRootType.CLASSES);
+    for (String url : urls) {
+      if (url.endsWith(DART_CORE_SUFFIX)) {
+        final String flutterUrl = url.substring(0, url.length() - DART_CORE_SUFFIX.length());
+        final String flutterPath = getPathForProjectLibraryUrl(flutterUrl, projectDir);
+        return flutterPath == null ? null : new FlutterSdk(flutterPath);
+      }
+    }
+    return null;
+  }
+
+  @Nullable
+  private static Library loadDartSdkLibrary(@NotNull VirtualFile projectDir) {
+    final VirtualFile libPath = projectDir.findFileByRelativePath(SDK_LIBRARY_PATH);
+    if (libPath == null) {
+      return null;
+    }
+
+    try {
+      final Element elt = JDOMUtil.load(new File(libPath.getPath()));
+      final ProjectLibraryTable table = new ProjectLibraryTable();
+      table.readExternal(elt);
+      return table.getLibraryByName("Dart SDK");
```

Comment:
I wonder if there's a more robust way to do this lookup than by string lookup.

cc @alexander-doroshko